application-task-manager-api/src/main/java/com/xwiki/taskmanager/internal/AbstractTaskEventListener.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xwiki.taskmanager.internal;
+
+import java.util.Arrays;
+import java.util.List;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.xwiki.model.reference.EntityReferenceResolver;
+import org.xwiki.model.reference.LocalDocumentReference;
+import org.xwiki.observation.AbstractEventListener;
+import org.xwiki.observation.event.Event;
+
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.doc.XWikiDocument;
+
+/**
+ * Base class for {@link TaskObjectUpdateEventListener} and {@link TaskMacroUpdateEventListener}.
+ *
+ * @version $Id$
+ * @since 1.0
+ */
+public abstract class AbstractTaskEventListener extends AbstractEventListener
+{
+    /**
+     * The name of the space of the application.
+     */
+    public static final String TASK_MANAGER_SPACE = "TaskManager";
+
+    /**
+     * The reference to the TaskClass page.
+     */
+    public static final LocalDocumentReference TASK_OBJECT_CLASS_REFERENCE =
+        new LocalDocumentReference(Arrays.asList(TASK_MANAGER_SPACE, "Code"), "TaskClass");
+
+
+    protected static final String TASK_UPDATE_FLAG = "taskUpdating";
+
+    @Inject
+    protected EntityReferenceResolver<String> resolver;
```

Comment:
You seem to resolve only document references in your listeners, so why not use a ``DocumentReferenceResolver``?